core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -36,13 +36,22 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
   private wordCloudElement: d3.Selection<SVGGElement, unknown, HTMLElement, any> | undefined;
   private c3ChartElement: c3.ChartAPI | undefined;
 
+  private updateSubscription: Subscription | undefined;
+
   constructor(
     private workflowStatusService: WorkflowStatusService
   ) {
   }
 
   ngAfterViewInit() {
+    // attempt to draw chart immediately
     this.drawChart();
+
+    // setup an event lister that re-draws the chart content every (n) miliseconds
+    // auditTime makes sure the first re-draw happens after (n) miliseconds has elapsed
+    this.updateSubscription = this.workflowStatusService.getResultUpdateStream().auditTime(2000).subscribe(update => {
```

Comment:
can this `2000` be declared as a static variable?

Reply:
done